bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/DefaultPerChannelBookieClientPool.java

```diff
@@ -63,12 +66,16 @@
         this.address = address;
         this.conf = conf;
 
-        this.shFactory = SecurityProviderFactoryFactory
-                .getSecurityProviderFactory(conf.getTLSProviderFactoryClass());
+        this.shFactory = SecurityProviderFactoryFactory.getSecurityProviderFactory(conf.getTLSProviderFactoryClass());
 
         this.clients = new PerChannelBookieClient[coreSize];
```

Comment:
Do we have to always create v2 clients even if I am using only v3 ?

Reply:
It doesn't. I should make it as optional. Fix it later